### DISCUSSION TITLE
Formula to compare 3 inputs to a truth table

### DISCUSSION THREAD

**Discussion by bigfoot53** [Budibase/budibase#9837] [Help]
Hello everyone, 
I am new to Budibase and trying to create a form to evaluate some performance data. I need to evaluate 3 inputs and compare the values to a truth table and then have the data "answer" populate in a new field called AA Total. I have tried the code below in the AA Total field and it doesn't show any errors  but also doesn't calculate :)  the 3 input fields are set as text becaus the input values are +, 0, or -  as shown in the truth table. 
Thanks for any help  for the newbie  :) 

```js
function calculateAATotal() { 
  // Get the input values from the user 
  var AAHigh = budibase.get("AA High input"); 
  var AAMedium = budibase.get("AA Medium input"); 
  var AALow = budibase.get("AA Low input"); 
 
  // Apply the truth table logic to calculate the AA Total value 
  var value = 0; 
  var input = AAHigh + AAMedium + AALow; 
 
  switch (input) { 
    case "+++": 
      value = 10; 
      break; 
    case "0++": 
      value = 9; 
      break; 
    case "-++": 
      value = 8; 
      break; 
    case "00+": 
      value = 7; 
      break; 
    case "-0+": 
      value = 6; 
      break; 
    case "000": 
      value = 5; 
      break; 
    case "--+": 
      value = 4; 
      break; 
    case "-00": 
      value = 3; 
      break; 
    case "--0": 
      value = 2; 
      break; 
    case "---": 
      value = 1; 
      break; 
    default: 
      console.log("Invalid input!"); 
      break; 
  } 
// Set the calculated AA Total value in the existing column 
  budibase.set("AA Total", value); 
} 
```

**Comment by melohagan** [comment]
Hey @bigfoot53 

Welcome to Budibase!

A couple of things to start off with. `budibase.get/set` is not a valid expression. If you are looking to get form input values, then you would want to use the form field bindings rather than formulas. 

These can be accessed by component's bindings that are nested within that form, and the bindings can be found in the panel on the left-hand side.
For example:

![Screenshot 2023-03-01 at 14 13 29](https://user-images.githubusercontent.com/101575380/222165291-5ac4e0b4-d358-446d-a276-df04925a537f.png)
![Screenshot 2023-03-01 at 14 18 13](https://user-images.githubusercontent.com/101575380/222166472-9d37eda5-e437-4e86-90ec-3bde239b2de4.png)

But if you wanted to use a JavaScript binding instead, you would click on the Lightning bolt icon and click on the JavaScript tab:

```js
function getName() { 
  // Get the input values from the user 
  var name = $("Form.Fields.Name");
 
  return name;
} 

return getName();
```

Note that JavaScript bindings must always return a value. So in this case I'm returning the output of the function.

![Screenshot 2023-03-01 at 14 21 19](https://user-images.githubusercontent.com/101575380/222167296-e50c2845-bd59-4804-b320-b37ea43f0286.png)

**Reply by bigfoot53** [reply]
thanks for the information,  i updated the code to this // Get the input values from the user 
  var AAHigh =$("AA High");
  var AAMedium =$("AA Medium");
	var AALow =$("AA Low");
  var value = 0;
	return AAHigh + AAMedium + AALow;
}	
return AATotal()

which returns the correct 3 inputs for aahigh, aamedium and aalow  How do i now get to use the truth table to change the results into the numerical value ?  I have tried a few variation with no results  :(

**Comment by bigfoot53** [comment]
With a little more review  it seems to  working with this code  :) 


function calculateAATotal() {
  // Get the input values from the user 
  var AAHigh = $("AA High");
  var AAMedium = $("AA Medium");
  var AALow = $("AA Low");
  var AATotal = $("AA Total")
  // Apply the truth table logic to calculate the AA Total value
  var value = 0;
  var input = AAHigh + AAMedium + AALow;  

  switch (input) {
    case "+++":
      value = 10;
      break;
    case "0++":
      value = 9;
      break;
    case "-++":
      value = 8;
      break;
    case "00+":
      value = 7;
      break;
    case "-0+":
      value = 6;
      break;
    case "000":
      value = 5;
      break;
    case "--+":
      value = 4;
      break;
    case "-00":
      value = 3;
      break;
    case "--0":
      value = 2;
      break;
    case "---":
      value = 1;
      break;
			
      }

  return AATotal, value;
}

return calculateAATotal();
